### DISCUSSION TITLE
Type Parameter in ViewModelBase.RefreshAsync

### DISCUSSION THREAD

**Discussion by b-higginbotham** [MarimerLLC/csla#1779] [General]
I am currently reviewing changes for a project that we will be upgrading from CSLA 4.7 to 5.3 in the next few weeks. This project uses the **ViewModelBase.DoRefresh** method of which has been deprecated and replaced with **ViewModelBase.RefreshAsync**. I'm reading through the source and the examples to get a better understanding of the impact this upgrade will have on this project and I don't understand the purpose of the type parameter in **ViewModelBase.RefreshAsync<F>**. It doesn't appear to be used by the **ViewModelBase** class itself, and the examples in the  ProjectTracker sample that override the method don't appear to do anything with it either.

What is the purpose of this type parameter? Are there any use cases in which would it need to be different from the type parameter **T** that is defined at the class level?

**Comment by rockfordlhotka** [comment]
I don't recall to be honest - need to review that code.

**Reply by rockfordlhotka** [reply]
@Hadronicus is this the one in `Csla.Xaml` or in `Csla.Blazor`?

**Reply by b-higginbotham** [reply]
I was looking at Csla.Xaml.
